### ISSUE TITLE
Warn when using namespace alias that isn't idiomatic according the Clojure Style Guide

### ISSUE THREAD

**Issue by sheluchin** [clj-kondo/clj-kondo#1839]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
The Clojure Style Guide has a [Use Idiomatic Namespace Aliases](https://github.com/bbatsov/clojure-style-guide#use-idiomatic-namespace-aliases) section which lists a set of namespace aliases. Adding a linter to clj-kondo to nudge developers towards adapting these community aliases in their own code would help increase the consistency of Clojure code bases.

**Describe the solution you'd like**
Add a linter that warns when using an unidiomatic namespace alias, where the idioms are defined in the Clojure Style Guide.

This can be implemented by adding a `:community-guide` flag to the [`:consistent-alias`](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#consistent-alias) linter. Namespaces which have explicit configuration in the `:consistent-alias` linter should overwrite the `:community-guide` configuration, in case users prefer _some_ aliases of their own, but generally want to adopt the community standards.

**Describe alternatives you've considered**
@borkdude suggested that alternatively there can be a new clj-kondo library created which would contain the community guide aliases so they could get merged to a project's clj-kondo config when imported. This option would require the minimal change to clj-kondo itself -- probably just documentation. On the down side, this approach may be difficult when installing clj-kondo as a binary instead of a project dependency. 

**Additional context**
Currently, the Style Guide recommends `clojure.spec.alpha` -> `spec` which is likely a bug. There is an open [PR](https://github.com/bbatsov/clojure-style-guide/pull/246) which changes this item and some others. There should be some expectation of the alias list evolving over time.

**Comment by sheluchin** [comment]
@borkdude went ahead and put up the https://github.com/clj-kondo/configs repo which includes the recently updated set of community guide aliases.

You can opt in by putting this in your project dependencies:
```clojure
io.github.clj-kondo/config-bbatsov-clojure-style-guide-aliases {:mvn/version "0.1.1"}
```

and then restarting clojure-lsp or running:
```bash
clj-kondo --lint $(clojure -Spath) --dependencies --skip-lint --copy-configs
```

Note that you can overwrite one more more of the entries by configuring your own [`:consistent-alias`](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md?rgh-link-date=2022-10-16T14%3A53%3A19Z#consistent-alias) linter.

Marking this as resolved as it seems to cover all the bases. Thank you, @borkdude!
